mteb/models/vllm_wrapper.py

```diff
@@ -0,0 +1,267 @@
+import atexit
+import logging
+from typing import TYPE_CHECKING, Any, Literal
+
+import numpy as np
+import torch
+from torch.utils.data import DataLoader
+
+from mteb._requires_package import requires_package
+from mteb.abstasks.task_metadata import TaskMetadata
+from mteb.models import ModelMeta
+from mteb.models.abs_encoder import AbsEncoder
+from mteb.types import Array, BatchedInput, PromptType
+
+if TYPE_CHECKING:
+    from vllm.config import PoolerConfig
+else:
+    PoolerConfig = Any
+
+logger = logging.getLogger(__name__)
+
+Dtype = Literal["half", "float16", "float", "float32", "bfloat16"]
+
+
+class VllmWrapperBase:
+    """Wrapper for vllm serving engine."""
+
+    convert = "auto"
```

Comment:
```suggestion
    convert = "auto"
    mteb_model_meta: ModelMeta | None = None
```
From discussion with @Samoed above